contracts/protocol/facets/ExchangeCommitFacet.sol

```diff
@@ -216,14 +192,193 @@ contract ExchangeCommitFacet is DisputeBase, BuyerBase, OfferBase, IBosonExchang
+                    _offerCreator,
+                    _fullOffer.offer.exchangeToken,
+                    offerCreatorAmount
+                );
+            }
+        }
+
+        if (_fullOffer.condition.method != BosonTypes.EvaluationMethod.None) {
+            if (_fullOffer.offer.creator == BosonTypes.OfferCreator.Seller) {
+                // validate seller parameters
+            } else {
+                addSellerParametersToBuyerOffer(_committer, offerId, _sellerParams);
+            }
+
+            commitToConditionalOffer(_committer, offerId, _conditionalTokenId);
+        } else {
+            if (_fullOffer.offer.creator == BosonTypes.OfferCreator.Seller) {
+                commitToOffer(_committer, offerId);
+            } else {
+                commitToBuyerOffer(offerId, _sellerParams);
+            }
+        }
+    }
+
+    /**
+     * @notice Verifies the offer and its signature.
+     *
+     * Reverts if:
+     * - Offer is not valid
+     * - Offer has been voided
+     * - Signature is invalid
+     *
+     * @param _fullOffer - the fully populated struct containing offer, offer dates, offer durations, dispute resolution parameters, condition, agent id and fee limit
+     * @param _offerCreator - the address of the offer creator
+     * @param _signature - the signature of the offer creator
+     */
+    function verifyOffer(
+        BosonTypes.FullOffer calldata _fullOffer,
+        address _offerCreator,
+        bytes calldata _signature
+    ) internal returns (bytes32 offerHash, uint256 offerId) {
+        // `_fullOffer.offer.voided` is checked in createOfferInternal
+        if (
+            _fullOffer.offer.id != 0 ||
+            _fullOffer.offer.royaltyInfo.length != 1 ||
+            _fullOffer.offer.priceType != BosonTypes.PriceType.Static
+        ) revert InvalidOffer();
+
+        offerHash = getOfferHash(_fullOffer);
+
+        // Check if the offer non-listed offer has been voided via `voidOffer`
+        // Does not apply to already listed offers, with `voided` set to true
+        offerId = protocolLookups().offerIdByHash[offerHash];
+        if (offerId == 0) {
+            if (_fullOffer.offer.creator == OfferCreator.Seller) {
+                // Validate caller is seller assistant
+                (, uint256 sellerId) = getSellerIdByAssistant(_offerCreator);
+                if (sellerId != _fullOffer.offer.sellerId) {
+                    revert NotAssistant();
+                }
+            } else if (_fullOffer.offer.creator == OfferCreator.Buyer) {
+                uint256 buyerId = getValidBuyer(payable(_offerCreator));
+                if (buyerId != _fullOffer.offer.buyerId) {
+                    revert NotBuyerWallet();
+                }
+            }
+            EIP712Lib.verify(_offerCreator, offerHash, _signature);
+        } else if (offerId == VOIDED_OFFER_ID) {
+            // Offer is voided
+            revert OfferHasBeenVoided();
+        }
```

Comment:
nothing in case the offer already exist?

Reply:
We just continue, this is to enable offers `quantity > 1`.
If it exists, I must have been validated already, so there's no need to verify the signatures. It also means we can skip another creation of a duplicate offer.  

If there are any other invalid parameters (e.g. offer expired, is sold out, etc), that's validated in the commit to offer step.